pkg/daemon/rpm-ostree.go

```diff
@@ -253,13 +253,28 @@ func (r *RpmOstreeClient) Rebase(imgURL, osImageContentDir string) (changed bool
 	return
 }
 
+func truncate(input string, length int) string {
+	asRunes := []rune(input)
+
+	if length > len(asRunes) {
+		length = len(asRunes)
+	}
+
+	return string(asRunes[:length])
+}
+
 // runGetOut executes a command, logging it, and return the stdout output.
 func runGetOut(command string, args ...string) ([]byte, error) {
 	glog.Infof("Running captured: %s %s", command, strings.Join(args, " "))
 	cmd := exec.Command(command, args...)
-	rawOut, err := cmd.CombinedOutput()
+	rawOut, err := cmd.Output()
 	if err != nil {
-		return nil, errors.Wrapf(err, "error running %s %s: %s", command, strings.Join(args, " "), string(rawOut))
+		errtext := ""
+		if e, ok := err.(*exec.ExitError); ok {
+			// Trim to max of 256 characters
+			errtext = fmt.Sprintf("\n%s", truncate(string(e.Stderr), 256))
```

Comment:
can we keep all error message. This will help to keep complete error messages captured when something fails.
Like, longer error message from one the bug we have seen:
```
E1113 00:46:05.603931    3636 writer.go:135] Marking Degraded due to: error running rpm-ostree override remove kernel kernel-core kernel-modules kernel-modules-extra --install kernel-rt-core --install kernel-rt-modules --install kernel-rt-modules-extra --install kernel-rt-kvm: Checking out tree 859d427...done
Enabled rpm-md repositories: coreos-extensions
rpm-md repo 'coreos-extensions' (cached); generated: 2020-11-11T18:27:37Z
Importing rpm-md...done
Resolving dependencies...done
error: Could not depsolve transaction; 4 problems detected:
 Problem 1: conflicting requests
  - nothing provides linux-firmware >= 20200619-99.git3890db36 needed by kernel-rt-core-4.18.0-240.1.1.rt7.55.el8_3.x86_64
 Problem 2: package kernel-rt-modules-extra-4.18.0-240.1.1.rt7.55.el8_3.x86_64 requires kernel-rt-uname-r = 4.18.0-240.1.1.rt7.55.el8_3.x86_64, but none of the providers can be installed
  - conflicting requests
  - nothing provides linux-firmware >= 20200619-99.git3890db36 needed by kernel-rt-core-4.18.0-240.1.1.rt7.55.el8_3.x86_64
 Problem 3: package kernel-rt-modules-4.18.0-240.1.1.rt7.55.el8_3.x86_64 requires kernel-rt-uname-r = 4.18.0-240.1.1.rt7.55.el8_3.x86_64, but none of the providers can be installed
  - conflicting requests
  - nothing provides linux-firmware >= 20200619-99.git3890db36 needed by kernel-rt-core-4.18.0-240.1.1.rt7.55.el8_3.x86_64
 Problem 4: package kernel-rt-kvm-4.18.0-240.1.1.rt7.55.el8_3.x86_64 requires kernel-rt = 4.18.0-240.1.1.rt7.55.el8_3, but none of the providers can be installed
  - conflicting requests
  - nothing provides linux-firmware >= 20200619-99.git3890db36 needed by kernel-rt-core-4.18.0-240.1.1.rt7.55.el8_3.x86_64
: exit status 1
```

Reply:
@sinnykumari (apologies for the potentially simple question) but wouldn't that error be contained/returned in rawOut (below)

Reply:
Hmm.  Right, so at 256 characters that would be truncated to something like:
```
error: Could not depsolve transaction; 4 problems detected:
 Problem 1: conflicting requests
  - nothing provides linux-firmware >= 20200619-99.git3890db36 needed by kernel-rt-core-4.18.0-240.1.1.rt7.55.el8_3.x86_64
```

Which isn't bad?

Actually though...why are we using `runGetOut()` for that?  Take this one for example:
https://github.com/openshift/machine-config-operator/blob/c415ce6aed25604bc1d2478951db16759dac31f6/pkg/daemon/update.go#L987
There we're gathering the stdout (and stderr) to a pipe, but just throwing it away.

I think perhaps the best is to pass through stdout (so regular information messages go to the MCD pod log) and capture stderr - write the whole stderr to our stderr, and also capture e.g. the last chunk of it here (which actually is similar to what this https://golang.org/pkg/os/exec/#ExitError is already doing, though it doesn't define "long" but from a look at current Go sources it seems to be 64k, which is actually wayyy too long to include in something like status text).

Reply:
I think it was just laziness that we have been reusing runGetOut() in both cases (just executing command and for parsing output). Thanks for creating https://github.com/openshift/machine-config-operator/pull/2587